code/lib/cli/src/automigrate/fixes/sveltekit-framework.test.ts

```diff
@@ -0,0 +1,173 @@
+        await expect(checkSvelteKitFramework({ packageJson, main })).rejects.toThrow();
+      });
+
+      it('with unsupported framework', async () => {
+        const packageJson = {
+          dependencies: {
+            '@sveltejs/kit': '^1.0.0-next.571',
+            '@storybook/svelte-vite': '^7.0.0',
+            '@storybook/html': '^7.0.0',
+          },
+        };
+        const main = {
+          framework: '@storybook/html',
+        };
+        await expect(checkSvelteKitFramework({ packageJson, main })).rejects.toThrow();
+      });
+
+      it('with unsupported framework+builder from SB 6.5', async () => {
+        const packageJson = {
+          dependencies: {
+            '@sveltejs/kit': '^1.0.0-next.571',
+            '@storybook/svelte-webpack5': '^7.0.0',
+            '@storybook/svelte': '^7.0.0',
+          },
+        };
+        const main = {
+          framework: '@storybook/svelte',
+          core: { builder: '@storybook/builder-webpack5' },
+        };
+        await expect(checkSvelteKitFramework({ packageJson, main })).rejects.toThrow();
+      });
+
+      it('with @storybook/svelte-webpack5 framework', async () => {
+        const packageJson = {
+          dependencies: {
+            '@storybook/svelte': '^7.0.0',
+            '@storybook/svelte-webpack5': '^7.0.0',
+            '@sveltejs/kit': '^1.0.0-next.571',
+          },
+        };
+        const main = {
+          framework: '@storybook/svelte-webpack5',
+        };
+        await expect(checkSvelteKitFramework({ packageJson, main })).rejects.toThrow();
+      });
+    });
+  });
+
+  describe('should migrate', () => {
+    it('from @storybook/svelte-vite', async () => {
+      const packageJson = {
+        dependencies: {
+          '@storybook/svelte': '^7.0.0',
+          '@storybook/svelte-vite': '^7.0.0',
+          '@sveltejs/kit': '^1.0.0-next.571',
+        },
+      };
+      const main = {
+        framework: '@storybook/svelte-vite',
+      };
+      await expect(checkSvelteKitFramework({ packageJson, main })).resolves.toMatchObject({
+        packageJson,
+        main: expect.objectContaining({}),
+        frameworkOptions: undefined,
+        dependenciesToRemove: ['@storybook/svelte-vite'],
+      });
+    });
+
+    it('from @storybook/svelte framework and @storybook/builder-vite builder', async () => {
+      const packageJson = {
+        dependencies: {
+          '@storybook/svelte': '^7.0.0',
+          '@storybook/builder-vite': '^7.0.0',
+          '@sveltejs/kit': '^1.0.0-next.571',
+        },
+      };
+      const main = {
+        framework: '@storybook/svelte',
+        core: { builder: '@storybook/builder-vite' },
+      };
+      await expect(checkSvelteKitFramework({ packageJson, main })).resolves.toMatchObject({
+        packageJson,
+        main: expect.objectContaining({}),
+        frameworkOptions: undefined,
+        dependenciesToRemove: ['@storybook/builder-vite', '@storybook/svelte'],
```

Comment:
I don't think we want to remove `@storybook/svelte-vite` in this case. Users will be directly referencing the renderer in their stories, so this will cause PNP/linting errors if we do. (I think)

Reply:
I think you're right. I was focusing on the types import, but I guess there's functionality that users import from the renderer directly.